### PR TITLE
HM-42 (Add ImageURL to AdditionalUserData) and HM-60 (small bugfix)

### PR DESCRIPTION
**Pull request by viktorgkw**

### CODE REVIEW

File: src/Images/Images.Api/Controllers/UserImagesController.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        IMediator mediator,
+        ILogger<ImageController> logger) : ControllerBase
+    {
+        private readonly IMediator _mediator = mediator;
+        private readonly ILogger<ImageController> _logger = logger;
+
+        [HttpPost]
+        [Route("{userId}")]
+        [Consumes("multipart/form-data")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```
Comment (rosteslav) [inline src/Images/Images.Api/Controllers/UserImagesController.cs]: Type!

File: src/Images/Images.Api/Controllers/UserImagesController.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                _logger.LogError("User image upload was not successful.");
+                return BadRequest();
+            }
+
+            return Ok(imageId);
+        }
+
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```
Comment (rosteslav) [inline src/Images/Images.Api/Controllers/UserImagesController.cs]: 200 of a what?

File: src/Images/Images.Api/Controllers/UserImagesController.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            var userId = User.Claims
+                .First(x => x.Type == ClaimTypes.Sid).Value;
+
+            await _mediator.Send(new DeleteUserImageCommand
+            {
+                UserId = userId
+            });
+
+            return Ok();
```
Comment (rosteslav) [inline src/Images/Images.Api/Controllers/UserImagesController.cs]: NoContent()

File: src/Images/Images.Application/Contracts/IImagesRepository.cs [hunk 4]
```diff
@@ -5,8 +5,10 @@ namespace BuildingMarket.Images.Application.Contracts
     public interface IImagesRepository
     {
         Task<IEnumerable<Image>> GetAllForProperty(int propertyId);
-        Task Add(Image image);
-        Task Delete(int imageId);
+        Task AddPropertyImage(Image image);
+        Task DeletePropertyImage(int imageId);
+        Task AddUserImage(string imageUrl, string userId);
```
Comment (rosteslav) [inline src/Images/Images.Application/Contracts/IImagesRepository.cs]: What's the difference, why AddPropertyImage is based on Image and AddUserImage is based on string?

File: src/Images/Images.Domain/Entities/AdditionalUserData.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace BuildingMarket.Images.Domain.Entities
+{
+    public class AdditionalUserData
+    {
+        [Key]
```
Comment (rosteslav) [inline src/Images/Images.Domain/Entities/AdditionalUserData.cs]: POCO Object - no logic here!

File: src/Images/Images.Domain/Entities/AdditionalUserData.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
+﻿using System.ComponentModel.DataAnnotations;
```
Comment (rosteslav) [inline src/Images/Images.Domain/Entities/AdditionalUserData.cs]: no usings here - clean

File: src/Images/Images.Infrastructure/Migrations/20231125173857_InitialCreate.cs
Comment (rosteslav) [inline src/Images/Images.Infrastructure/Migrations/20231125173857_InitialCreate.cs:1]: Don't modify auto-generated file

File: src/Images/Images.Infrastructure/Persistence/ImagesDbContext.cs [hunk 7]
```diff
@@ -105,6 +106,38 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                     .HasColumnType("timestamptz")
                     .HasDefaultValue(new DateTime(1, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc));
             });
+
+            modelBuilder.Entity<AdditionalUserData>(entity =>
```
Comment (rosteslav) [inline src/Images/Images.Infrastructure/Persistence/ImagesDbContext.cs:110]: This is not owned here -> no migrations here
This is owned in Auth microservice

**Review by rosteslav — CHANGES_REQUESTED** [review]

File: src/Images/Images.Application/Contracts/IImagesRepository.cs (hunk 4 above)
Comment (viktorgkw) [reply src/Images/Images.Application/Contracts/IImagesRepository.cs]: AddPropertyImage accepts Image that has PropertyId and ImageURL which then adds a new record in the Images table. On the other side AddUserImage updates the ImageURL in the user's additionaluserdata data.

```
await _context.AdditionalUserData
        .Where(addData => addData.UserId == userId)
        .ExecuteUpdateAsync(p => p
            .SetProperty(data => data.ImageURL, data => imageUrl));
```

So it doesn't add a record in the Image table.

File: src/Images/Images.Infrastructure/Migrations/20231125173857_InitialCreate.cs
Comment (rosteslav) [inline src/Images/Images.Infrastructure/Migrations/20231125173857_InitialCreate.cs:1]: Don't touch auto generated files :) Sory if I got confused

File: src/Images/Images.Infrastructure/Persistence/ImagesDbContext.cs [hunk 8]
```diff
@@ -105,6 +106,38 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+            {
+                entity.ToTable("AdditionalData", "security");
```
Comment (rosteslav) [inline src/Images/Images.Infrastructure/Persistence/ImagesDbContext.cs]: This is not handled here! add like here:
entity.ToTable("Images", "properties", i => i.ExcludeFromMigrations());

File: src/Auth/Auth.Domain/Entities/AdditionalUserData.cs
Comment (rosteslav) [inline src/Auth/Auth.Domain/Entities/AdditionalUserData.cs:1]: No         [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
- keep it pure

File: src/Images/Images.Application/Contracts/IImagesRepository.cs (hunk 4 above)
Comment (rosteslav) [reply src/Images/Images.Application/Contracts/IImagesRepository.cs]: I see now. Ok, separate this repository to two - one for the profile images and one for the property images. There you should keep names as Add, Delete, etc. ImgbbService stays one, for sure

**Review by rosteslav — CHANGES_REQUESTED** [review]

**Review by rosteslav — APPROVED** [review]

**Pull request merged**